staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -82,10 +82,21 @@ func validator(s *schema.Structural, isResourceRoot bool, declType *model.DeclTy
 	if len(s.Properties) > 0 {
 		propertiesValidators = make(map[string]Validator, len(s.Properties))
 		for k, prop := range s.Properties {
-			if f, ok := declType.Fields[k]; ok {
-				if p := validator(&prop, prop.XEmbeddedResource, f.Type, perCallLimit); p != nil {
-					propertiesValidators[k] = *p
+			var fieldType *model.DeclType
+			if escapedPropName, ok := model.Escape(k); ok {
+				if f, ok := declType.Fields[escapedPropName]; ok {
+					fieldType = f.Type
+				} else {
+					// fields with unknown types are omitted from CEL validation entirely
+					continue
 				}
+			} else {
+				// field may be absent from declType if the property name is unescapable, in which case we should convert
+				// the field value type to a DeclType.
+				fieldType = model.SchemaDeclType(&prop, prop.XEmbeddedResource)
```

Comment:
is it possible for this to return nil? if so, should we `continue`?

Reply:
It is possible. When we skip CEL validation if the OpenAPIv3 is invalid it should stop happening, but I'd like to have the safety checks regardless. Fixed.